src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1214,83 +1220,113 @@ public void downloadMailArchiveZipFile(Environment env, HttpServletRequest reque
+                channelId = channelId.substring(("/mailarchive/channel/" + site_id + "/").length());
+                String channelName = channel.getString("displayTitle");
+
+                if (folderForChannels)
+                {
+                    ZipEntry folderEntry = new ZipEntry(channelName + "/");
+                    try {
+                        out.putNextEntry(folderEntry);
+                    } catch (IOException e) {
+                        String ioError = "downloadMailArchiveZipFile: problem adding zip folder for MailArchive channel "
+                                + channelName + " " + e;
+                        log.error(ioError);
+                    }
+                }
+
+                // get all email messages in the channel
+                requestUrl = env.getProperty(Utils.ENV_PROPERTY_CTOOLS_SERVER_URL)
+                        + "direct/mailarchive/channelMessages/" + site_id + "/" + channelId + ".json?_sessionId="
+                        + sessionId;
+                JSONObject messagesJSON = new JSONObject(restTemplate.getForObject(requestUrl,
+                        String.class));
+                JSONArray messages = messagesJSON.getJSONArray(Utils.JSON_ATTR_MAILARCHIVE_COLLECTION);
+                Map<String, String[]> parameterMap = request.getParameterMap();
+                String destination_type = parameterMap.get("destination_type")[0];
+                JSONObject messageStatus = new JSONObject();
+
+                if(destination_type.equals("mailarchivezip")) {
+                    for (int iMessage = 0; iMessage < messages.length(); iMessage++) {
+                        JSONObject message = messages.getJSONObject(iMessage);
+
+                        // create file for each message
+                        String messageFolderName = getMailArchiveMessageFolderName(message, channelName, folderForChannels);
+
+                        messageStatus.put(Utils.JSON_ATTR_MAIL_MESSAGE, messageFolderName);
+
+                        // 1. write the message file
+                        messageStatus = handleMailArchiveMessage(out, message, messageFolderName,
+                                messageStatus);
+
+                        // 2. get attachments, if any
+                        messageStatus = handleMailArchiveMessageAttachments(
+                                sessionId, httpContext, out, message, messageFolderName, messageStatus);
+
+                        messagesStatus.put(messageStatus);
+                    }
+                }
+                else if(destination_type.equals("mailarchivembox")) {
+                    String messageFolderName = getMailArchiveMboxMessageFolderName(site_id, channelName, folderForChannels);
+                    StringBuilder msgBundle = new StringBuilder();
+                    for (int iMessage = 0; iMessage < messages.length(); iMessage++) {
+                        JSONObject message = messages.getJSONObject(iMessage);
```

Comment:
dave, IMHO generalizing attachment handling for both Zip and mbox is not possible.  The way mbox is handling it is it is giving it to the java mail library and it doing the base64 encoding of attachment. The mail Zip workflow the attachment handling is done through the Java Zip library. They both are working in different way. Generalizing it will make it complicated and time consuming at this point of time